docs/proposal/performance_monitoring.md

```diff
@@ -0,0 +1,82 @@
+---
+title: proposal of Performance Monitoring
+authors:
+- "@skwwwwww" # Authors' GitHub accounts here.
+reviewers:
+- ""
+- TBD
+approvers:
+- ""
+- TBD
+
+creation-date: 2024-09-21
+
+---
+
+## proposal of Kmesh observability
+
+<!--
+This is the title of your KEP. Keep it short, simple, and descriptive. A good
+title can help communicate what the KEP is and should be considered as part of
+any review.
+-->
+### Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap.
+
+A good summary is probably at least a paragraph in length.
+-->
+
+The significance of performance monitoring in Kmesh as the foundation for efficient, scalable, and robust grid systems is paramount. In Kmesh, monitoring key metrics such as Kmesh_deamon activity, the number of eBPF maps, the number of entries in each map, and the duration of eBPF method executions is crucial to ensure optimal system performance.
+
+In this proposal, I will analyze the performance monitoring metrics of Kmesh. I will also suggest the implementation of enhanced observability features within Kmesh to capture these critical performance indicators. This will allow users to seamlessly monitor Kmesh's performance and ensure system efficiency.
+
+### Proposal
+
+Kmesh needs to collect metrics through the kernel and pass them on to the user mode. In the user mode, The data related to eBPF maps and operation duration can ultimately be queried through Prometheus and visualized using Grafana.
+
+#### Design Details
+
+##### metrics related operation duration
+
+This is because Kmesh needs to get metrics from the kernel and sent them to the user mode. We need a bpf map to record the metrics, as a vehicle for transferring.
+
+So we need to define a bpf map that contains all the required metrics:
+
+```
+struct operation_usage_data {
+    __u64 start_time;
+    __u64 end_time;
+    __u32 operation_type;
+};
+
+struct operation_usage_key {
+    __u32 tid;
```

Comment:
is this pid?

Reply:
how can we map it to a pod

Reply:
The tid is the unique identifier of a thread obtained through bpf_get_current_pid_tgid().

A unique kmesh-daemon runs on each node, and kmesh collects data from different nodes, allowing the data to be associated with the respective pods.